azurerm/resource_arm_kubernetes_cluster_test.go

```diff
@@ -0,0 +1,189 @@
+package azurerm
+
+import (
+	"fmt"
+	"net/http"
+	"os"
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAzureRMKubernetesCluster_basic(t *testing.T) {
+	ri := acctest.RandInt()
+	clientId := os.Getenv("ARM_CLIENT_ID")
+	clientSecret := os.Getenv("ARM_CLIENT_SECRET")
+	config := testAccAzureRMKubernetesCluster_basic(ri, clientId, clientSecret, testLocation())
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMKubernetesClusterDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMKubernetesClusterExists("azurerm_kubernetes_cluster.test"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAzureRMKubernetesCluster_complete(t *testing.T) {
```

Comment:
it'd be good to add tests covering both adding an agent pool and upgrading the version of Kubernetes - we can do this using multiple test steps

Reply:
added tests for upgrade and agent count, let me know what you think